### ISSUE TITLE
replace uses of distributions registry-client because it's deprecated and removed

### ISSUE THREAD

**Issue by thaJeztah** [moby/moby#46940]
### Description

relates to

- https://github.com/distribution/distribution/pull/4126
- https://github.com/distribution/distribution/issues/4119

The distribution's client was moved to an internal package, and docker/distribution ("v2") is about to be deprecated in favour of distribution/distribution "v3".

We need to replace uses of the client package in our code (we can temporarily use an internal fork if needed), and migrate to github.com/distribution/distribution v3 for the remaining parts we need (if any).

**Comment by corhere** [comment]
What's the rush? v2 isn't going to disappear when v3.0.0 gets tagged, so we don't have to change anything unless bugs come up that need fixing. And our use of the distribution client is legacy anyway: the containerd backend uses containerd's distribution client. We will be able to drop the dependency much more easily when we get rid of the graphdrivers. If bugs in the distribution client come up which need fixing before then, I think we should copy the client code into the repo, like we did with libnetwork. Carrying an internal fork in a separate repo would be just as much of an ongoing pain as merging fixes to the v2 branch of `distribution/distribution`.

**Comment by thaJeztah** [comment]
> What's the rush? v2 isn't going to disappear when v3.0.0 gets tagged, so we don't have to change anything unless bugs come up that need fixing

Yes, I think there's  quite a strong urge from the distribution maintainers to EOL v2 when v3 is released. There's already a (rather unfortunate) but in v2 (introduced by yours truly) that has a fix merged, but it's uncertain if another v2.x release will see light of day; https://github.com/distribution/distribution/pull/4114

From that perspective I considered it would be better to "own" the client, so that we have control (specifically for LTS branches), or at least it could make life easer in cases where there _is_ a bug to fix, and we either need to fork "under pressure" at that moment.
